client/src/components/Modals/EditTeam/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { ReactElement, FC, useState } from 'react';
+import {
+  Button,
+  ButtonProps,
+  Form,
+  Icon,
+  Input,
+  Modal,
+  ModalProps,
+} from 'semantic-ui-react';
+import Swal from 'sweetalert2';
+
+import { ITeam } from 'ssw-common';
+
+import './styles.scss';
+import { useTeams } from '../../../contexts';
+import { createManyTeams, updateManyTeams } from '../../../api';
+import { useEffect } from 'react';
+
+// interface TeamModalProps extends ModalProps {}
+
+const TeamModalTrigger: FC<ButtonProps> = ({ ...rest }): ReactElement => (
+  <Button {...rest} icon>
+    <Icon name="pencil" />
+  </Button>
+);
+
+// const copy = [...teams.map((t) => ({ ...t }))];
+
+const TeamModal: FC<ModalProps> = ({ ...rest }): ReactElement => {
+  const { teams, fetchTeams } = useTeams();
+  const [isOpen, setIsOpen] = useState(false);
+  const [formValues, setFormValues] = useState<ITeam[]>([]);
+
+  const [clonedTeams, setClonedTeams] = useState<Partial<ITeam>[]>([]);
+
+  useEffect(() => {
+    fetchTeams();
+    setFormValues(teams);
+
+    const clone = teams.map((team) => ({
+      _id: team._id,
+      name: team.name,
+      color: team.color,
+      active: team.active,
+    }));
+
+    setClonedTeams([...clone]);
+    console.log(teams);
+  }, [isOpen]);
+
+  // Add a new field in form
+  const addField = () => {
+    setFormValues([
+      ...formValues,
+      { _id: 'NEW', name: '', color: '#3d4f91', active: true },
+    ]);
+  };
+
+  // Remove a field in form
+  const removeField = (i: number) => {
+    const newFormValues = [...formValues];
+    newFormValues.splice(i, 1);
+    setFormValues(newFormValues);
+  };
+
+  // Update team names (input type = text)
+  const updateFieldText = (i: number, text: string) => {
+    const formValue = formValues[i];
+    formValue.name = text;
+    setFormValues([...formValues]);
+  };
+
+  // Update team colors (input type = color)
+  const updateFieldColor = (i: number, color: string) => {
+    const formValue = formValues[i];
+    formValue.color = color;
+    setFormValues([...formValues]);
+  };
+
+  // Save newTeams and changedTeams
+  const saveForm = (): void => {
+    const newTeams = formValues.filter((team) => team._id === 'NEW');
+    const parsedNewTeams: Partial<ITeam>[] = newTeams.map(
+      ({ _id, ...rest }) => rest,
+    );
+
+    const changedTeams: ITeam[] = [];
+
+    clonedTeams.forEach((team, index) => {
+      if (
+        team.name !== formValues[index].name ||
+        team.color !== formValues[index].color
+      ) {
+        changedTeams.push(formValues[index]);
+      }
+    });
+
+    // Check if the changedTeam names is in cloned Teams
+    // Check if the new team names
+    const valuesName = formValues.map((item) => item.name);
+    const isDuplicate = valuesName.some(
```

Comment:
Stuff liking checking duplicates and removing the fields that don't belong should be handled in helper functions.